docs/DeveloperGuide.md

```diff
@@ -1,25 +1,283 @@
 # Developer Guide
 
 ## Design & implementation
+<!-- @@author iamchenjiajun -->
+### Architecture
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+##### Architecture Diagram
+Below is an architecture diagram of termiNus.
+
+![ArchitectureDiagram](./images/ArchitectureDiagram.png)
+
+- `Duke` is the main object of the program and handles all the logic and models related to the program.
+- `Ui` is the main object that provides an interface between `Duke` and the user.
+- `Command` represents a command that is provided by the user. `Ui` reads the command and it is sent to `Parser` to create a new `Command` object.
+- `Duke` executes the command and shows the output to the user using `Ui`.
+- `Command` object modifies the state of `ItemList`, which consists of multiple lists for different types of `Items`.
+- `Storage` takes the state of `ItemList` and stores it to file.
+
+##### Sequence Diagram
+Below is a sequence diagram of how the main program functions.
+
+![DukeSequenceDiagram](./images/DukeSequenceDiagram.png)
```

Comment:
This diagram is inconsistent with the next diagram about storage:
![image](https://user-images.githubusercontent.com/26886705/97539576-2a2dd880-19fd-11eb-899c-22bb88de0f83.png)
This diagram shows that save() is executed repeatedly but Storage is a singleton in this process.
![image](https://user-images.githubusercontent.com/26886705/97539661-521d3c00-19fd-11eb-83e4-40a683b5e3bb.png)
However, this diagram shows Storage object being distroyed after one call to save().
Is one of the diagrams wrong?